### PR TITLE
Complete Project

### PR DESCRIPTION
**Pull request by niicode**
In this PR, I implemented the cash app webapp with the following features:
[Live presentation](https://drive.google.com/file/d/1m-6LjyFIBhzupzgyQrEDKx05gTpfj1Wq/view?usp=sharing)

Login, sign up, and splash screen
Category Page
Transaction page
Wrote test for the app

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,237 @@
...
+
+## ⭐️ Show your support <a name="support"></a>
+
+If you like this project buy the team a [drink](.../.../)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
```
Comment (SajjadAhmad14) [inline README.md:178]: Please give appropriate credit to `Gregoire Vella` on Behance, the author of the original design in `Acknowledgements` section of README.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,237 @@
...
+
+- **Create a category**
+- **Add a transaction**
+- **Sign up and log in page**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+
+## 🚀 Live Demo <a name="live-demo"></a>
```
Comment (SajjadAhmad14) [inline README.md:72]: Please deploy the app online and add a link to it in README.

File: app/views/categories/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<%= render 'common/navigation', :title => 'Categories' %>
```
Comment (SajjadAhmad14) [inline app/views/categories/index.html.erb:1]: Please follow the same layout for every category card on `categories` page as in the design. Moreover, please use the background color for `Add new category` button from the design page.

File: app/views/categories/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        <div class="item-details flex">
+          <%= link_to category_transactions_path(category_id: category.id), id: "Group-#{category.id}" do %>
+            <h3> <%= category.name %> </h3>
+          <% end %>
+          <p> $<%= category.total_amount %> </p>
+        </div>
+      </li>
+    <% end %>
+  </ul>
+  <%= link_to 'Add New Category', new_category_path,  class: "btn" %>
```
Comment (SajjadAhmad14) [inline app/views/categories/index.html.erb:17]: Please make `Add new category` button fixed in the bottom of the page as per design requirements.

File: app/views/home/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<section class="greeting container flex">
```
Comment (SajjadAhmad14) [inline app/views/home/index.html.erb:1]: Please use `Proxima Nova Light` font for general text and `Proxima Nova Bold` for headings as defined in the design page in the whole app.

File: app/views/transactions/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<%= render 'common/navigation', :title => 'Transactions' %>
```
Comment (SajjadAhmad14) [inline app/views/transactions/index.html.erb:1]: Please add a "Back" button (<) on `Transactions` page so when the user clicks it, it navigates the user to the home page.

File: app/views/categories/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<%= render 'common/navigation', :title => 'Categories' %>
+<section class="access flex">
+  <%= form_with model: group, url: categories_url, method: :post do |f| %>
```
Comment (SajjadAhmad14) [inline app/views/categories/new.html.erb:3]: Please add a "Back" button (<) on `Add a new category` page so when the user clicks it, it navigates the user to the home page.

File: app/views/transactions/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,23 @@
+<%= render 'common/navigation', :title => 'Transaction' %>
+<section class="access flex">
+  <%= form_with model: operation, url: category_transactions_url(category_id: id), method: :post do |f| %>
```
Comment (SajjadAhmad14) [inline app/views/transactions/new.html.erb:3]: Please add a "Back" button (<) on `Add a new transaction` page so when the user clicks it, it navigates the user to the home page.

File: spec/models/group_spec.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe Group, type: :model do
+  let(:user) { User.create(name: 'Blastus cardif', email: 'test@test.com', password: 'password') }
+  let(:group) { Group.create(name: 'new watch', icon: 'https://i.imgur.com/Ar3Lf3Dt.png', user_id: user.id) }
+
```
Comment (SajjadAhmad14) [inline spec/models/group_spec.rb:6]: Please add unit test for `ordered_transactions` from Group model

File: spec/models/user_spec.rb [hunk 10]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
Comment (SajjadAhmad14) [inline spec/models/user_spec.rb:3]: Please add unit tests for `ordered_groups` in User model.

File: spec/rails_helper.rb [hunk 11]
```diff
@@ -0,0 +1,63 @@
+# This file is copied to spec/ when you run 'rails generate rspec:install'
+require 'spec_helper'
```
Comment (SajjadAhmad14) [inline spec/rails_helper.rb:2]: Please add integeration specs for all the important app components.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Nii, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
App UI is facianating. :heavy_check_mark:
PR has no linter errors. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md (hunk 2 above)
Comment (niicode) [reply README.md:72]: @SajjadAhmad14 heroku has no free plans so I am not able to deploy there, I tried deploying on Render but it has worked for me

File: README.md [hunk 12]
```diff
@@ -1,24 +1,206 @@
...
+
+- register and log in, so that the data is private to them.
+- introduce new transactions associated with a category.
+- see the money spent on each category.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
```
Comment (CollinsTatang) [inline README.md:72]: Hi @niicode 
You have done a great job implementing some of the changes requested by the previous reviewer.

- [ ] Please you will have to deploy your app online and add the link to your README file to fully complete the requirement of this Capstone Project.

 -  Please follow the step in this [article](https://render.com/docs/deploy-rails) and check out this [Repo](https://github.com/CollinsTatang/helpjuice-project/blob/develop/render.yaml) for little changes to your `render.yaml` file.
 ### Please pay close attention your app will be deploy successfully

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @niicode, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @niicode,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**